src/push.rs

```diff
@@ -215,6 +223,16 @@ pub async fn build_profile(data: PushProfileData<'_>) -> Result<(), PushProfileE
         &data.deploy_data.profile.profile_settings.path
     );
 
+    let nix_version_output = Command::new("nix").arg("--version")
+        .output()
+        .await.map_err(PushProfileError::NixVersion)?;
+
+    let version_regex = Regex::new(r"(\d+\.\d+\..+)").unwrap();
+    let nix_version = version_regex.
```

Comment:
"idiomatic" rust code keeps the `.` at the beginning of the next line as in

```
let nix_version = version_regex
    .find(std::str::from_utf8(&nix_version_output.stdout)
        .map_err(PushProfileError::NixVersionUtf8)?)
    .map(|v : regex::Match| -> &str { v.as_str() })
    .ok_or(PushProfileError::NixVersionParse)?;
```

maybe we should introduce some pre-commit hooks for deploy-rs or at least a formatter via flakes, those could also be integrated into the ci